testsrc/README.md

```diff
@@ -32,14 +35,74 @@ This behavior can be changed through different means:
 
 ## Running a specific TestSuite
 ```
-./gradlew test --tests org.mozilla.javascript.tests.Test262SuiteTest 
+./gradlew test --tests org.mozilla.javascript.tests.Test262SuiteTest
+```
+
+## test262.properties format
+The [test262.properties](test262.properties) file is used to specify which tests from the official ECMAScript Test Suite to include/exclude,
+so that the test suite can pass even though Rhino is not yet 100% standards compliant
+
+The [test262.properties](test262.properties) file:
+- lists the subfolders of the [test262](../test262) folder to include or exclude when running tests
+- lists the .js test files that are expected to fail per (sub)folder
+
+**Basics**
+```
+built-ins/Array <- include all the tests in the test262/built-ins/Array folder
+    from/calling-from-valid-1-noStrict.js <- but expect that this tests fails
+```
+If `built-ins/Array/from/calling-from-valid-1-noStrict.js` indeed fails, it wont fail the test suite. If it passes, this will be logged while running the test suite
+
+**Skipping entire folders**
+```
+~built-ins/decodeURI
+    name.js
+    S15.1.3.1_A2.4_T1.js
+    S15.1.3.1_A5.2.js
+```
+Folders can be prefixed with a `~` to mark the folder to be skipped entirely, for example because it contains all tests for a feature not yet supported by Rhino or because running the tests take a long time.
+Any files listed for a skipped folder will be skipped as well.
+
+**Expecting all files in a (sub)folder to fail**
+```
+built-ins/Array
+    *prototype/flatMap
 ```
+If all files in a (sub) folder are expected to fail, instead of listing all files explicitly, the folder can be marked as expected to contain only failing tests by prefixing the folder path with an `*`
```

Comment:
A syntax that folks might understand better would be "prototype/flatMap/*". Or even use "**" to indicate multiple levels of directory. Is that a big change -- it might require creating a regex.

Reply:
Fairly easy to change and I had thought about it, but I chose to go with a prefix instead of a suffix because if you glance over the file, the prefix stands out more that a suffix would.

But if people feel a suffix makes more logical sense, I can change it, np

Reply:
Not sure how much weight I have, but I vote to leave it as a prefix for the reason mentioned.

Reply:
I vote for suffix

Reply:
So I guess @gbrail will be the deciding vote?

Reply:
A `~` prefix already means to skip a folder. Could that be used universally to skip a subfolder or a single file within a folder?

Reply:
Skipping using ~ is different from marking a (sub)folder to contain only failing tests: skipping means the tests aren't run at all. Marked as failing means the tests are run, but don't fail the test suite if they indeed fail

Reply:
My mistake, but I guess the question still holds. Is there a way to skip a subset of a folder? I have no idea if this is really necessary, and it may not be.

Do you even need a symbol to mark a subfolder as failing? The files don't have one, and the subfolder is listed in the same way, correct? Does the symbol in this case only serve to identify it as a folder instead of a file?

Reply:
> Do you even need a symbol to mark a subfolder as failing? The files don't have one, and the subfolder is listed in the same way, correct? Does the symbol in this case only serve to identify it as a folder instead of a file?

True, didn't think of that. So we have the following options to choose from then:

```
built-ins/Array
    *prototype/flatMap 21/21 (100.0%)
```
```
built-ins/Array
    prototype/flatMap 21/21 (100.0%)
```
```
built-ins/Array
    prototype/flatMap/ 21/21 (100.0%)
```
```
built-ins/Array
    prototype/flatMap/* 21/21 (100.0%)
```
```
built-ins/Array
    prototype/flatMap/** 21/21 (100.0%)
```
> My mistake, but I guess the question still holds. Is there a way to skip a subset of a folder? I have no idea if this is really necessary, and it may not be.

ATM there are no subfolders or .js files that are being skipped in test262.properties, only what I call topLevel folders (direct children of test262/test/). Had a look and concluded that attempting to skip subfolders or individual .js files under those would currently cause issues, both in running the test suite and in regenerating the .properties file. So for now I went ahead and made sure skipping folders is only supported for topLevel folders. If the need arises in the future to be able to skip on a lower level, that could be added. Note that most subfolders that only contain failing tests are related to unsupported features, which, if properly tagged as such in test262 and properly excluded by us (we have a hardcoded list of unsupported features), they will be skipped anyway.

Reply:
I would just go with the `!nothing` option. A prefix is not necessary. There is already a de facto "suffix" because you can tell by the absence of `.js` at the end that it is a folder, plus we don't have trailing slashes for the top level folders.

Will the generator create these folder exceptions instead of listing each file individually, or is this only a manual action?

Reply:
I'm ok with the 'nothing'  if everyone else agrees with that as well

If the generator concludes that all files in a folder (that is not marked to be skipped) fail, it'll just include the folder and not the individual files (unless you not enable 'rollup' through the command line (by default 'rollup' is enabled

Reply:
The trailing / is also an option for me, maybe a bit more readable.

Adding a trailing / to the topLevel folder entries for consistency wouldn't be difficult